test/e2e/prometheus_test.go

```diff
@@ -758,50 +705,52 @@ func testPromRemoteWriteWithTLS(t *testing.T) {
 			framework.SetupPrometheusRBAC(context.Background(), t, testCtx, ns)
 			name := "test"
 
-			// apply authorized certificate and key to k8s as a Secret
-			createK8sResources(t, ns, certsDir, test.ClientKey, test.ClientCert, test.CA)
+			// Create the secrets/configmaps storing the TLS certificates.
+			err := framework.CreateCertificateResources(ns, certsDir, tc.rwConfig)
+			if err != nil {
+				t.Fatal(err)
+			}
 
-			// Setup a sample-app which supports mTLS therefore will play 2 roles:
-			// 	1. app scraped by prometheus
-			// 	2. TLS receiver for prometheus remoteWrite
-			createK8sSampleApp(t, name, ns)
+			if err = deployInstrumentedApplicationWithTLS(name, ns); err != nil {
+				t.Fatal(err)
+			}
 
-			// Setup monitoring.
-			prometheusCRD, prometheusRecieverSvc, err := createK8sAppMonitoring(name, ns, test)
+			prometheus, svc, receiver, receiverSvc, err := createRemoteWriteStack(name, ns, tc.rwConfig)
 			if err != nil {
 				t.Fatal(err)
 			}
 
-			// Check for proper scraping.
-			promSVC := framework.MakePrometheusService(name, name, v1.ServiceTypeClusterIP)
-			if err := framework.WaitForHealthyTargets(context.Background(), ns, promSVC.Name, 1); err != nil {
-				framework.PrintPrometheusLogs(context.Background(), t, prometheusCRD)
+			// Wait for the instrumented application to be scraped.
+			if err := framework.WaitForHealthyTargets(context.Background(), ns, svc.Name, 1); err != nil {
+				framework.PrintPrometheusLogs(context.Background(), t, prometheus)
+				time.Sleep(time.Hour)
 				t.Fatal(err)
 			}
 
-			//TODO: make it wait by poll, there are some examples in other tests
-			// use wait.Poll() in k8s.io/apimachinery@v0.18.3/pkg/util/wait/wait.go
-			time.Sleep(45 * time.Second)
```

Comment:
Removing this 45s sleep reduces the test duration significantly (from ~45m to ~25m).